RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,277 @@
+from RUFAS.current_day_conditions import CurrentDayConditions
+from RUFAS.data_structures.animal_to_manure_connection import ManureStream
+from RUFAS.biophysical.manure.manure_constants import ManureConstants
+from RUFAS.output_manager import OutputManager
+from RUFAS.time import Time
+from RUFAS.units import MeasurementUnits
+
+
+class Separator(Processor):
+    """
+    A manure processor that separates a portion of solids from manure.
+
+    Parameters
+    ----------
+    name : str = ""
+        The name of the separator.
+    percent_dry_solids : float = 0.0
+        The dry matter content (percent DM) of separated manure solids.
+    ammoniacal_nitrogen_efficiency : float = 0.0
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float = 0.0
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float = 0.0
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float = 0.0
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float = 0.0
+        The efficiency of the separator in removing ash from the manure.
+    volatile_solids_efficiency : float = 0.0
+        The efficiency of the separator in removing volatile solids from the manure.
+    total_solids_efficiency : float = 0.0
+        The efficiency of the separator in removing total solids from the manure.
+
+    Attributes
+    ----------
+    is_housing_emissions_calculator : bool = False
+        A flag to indicate if the processor is a housing emissions calculator. Will always be false for a separator.
+    name : str
+        The name of the separator.
+    prefix : str
+        The prefix to be used for naming output variables.
+    percent_dry_solids : float
+        The dry matter content (percent DM) of separated manure solids.
+    ammoniacal_nitrogen_efficiency : float
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float
+        The efficiency of the separator in removing ash from the manure.
+    volatile_solids_efficiency : float
+        The efficiency of the separator in removing volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    om : OutputManager
+        The output manager instance used to store and manage output data.
+
+    """
+
+    def __init__(
+        self,
+        name: str = "",
```

Comment:
```suggestion
        name: str,
```
The name should _never_ be blank, it should always be defined by the user. If it is blank, it could disrupt the operations of `ManureManager` in a catastrophic way and the make outputs either really difficult or impossible to interpret.